pages/api/applications/[applicationId]/stages/index.ts

```diff
@@ -1,5 +1,133 @@
+  },
+  [MessageType.INVALID_DATE]: {
+    type: ERROR,
+    message: 'Application Stage date is invalid.',
+  },
+  [MessageType.INVALID_APPLICATION_ID]: {
+    type: ERROR,
+    message: 'Query parameter applicationId (.../applications/:applicationId/...) is invalid.',
+  },
+  [MessageType.APPLICATION_NOT_FOUND]: {
+    type: ERROR,
+    message: 'User does not have any application with the given ID.',
+  },
+  [MessageType.CREATED]: { type: SUCCESS, message: `Application stage has been created.` },
+});
+
+async function handler(
+  uid: string,
+  req: NextApiRequest,
+  res: NextApiResponse<ApiResponse<ApplicationStageData | ApplicationDataWithStages>>,
+) {
+  const errorMessageIfInvalid = validatePathParameters(req);
+  if (errorMessageIfInvalid !== null) {
+    return res.status(HttpStatus.BAD_REQUEST).json(createJsonResponse({}, messages[errorMessageIfInvalid]));
+  }
+
+  const application = await getApplication(uid, Number(req.query.applicationId));
+  if (application === null) {
+    return res.status(HttpStatus.NOT_FOUND).json(createJsonResponse({}, messages[MessageType.APPLICATION_NOT_FOUND]));
+  }
+
+  switch (req.method) {
+    case HttpMethod.POST:
+      return postHandler(application, req, res);
+    case HttpMethod.GET:
+      return getHandler(application, res);
+    default:
+      return rejectHttpMethod(res, req.method);
+  }
+}
+
+async function postHandler(
+  application: ApplicationDataWithStages,
+  req: NextApiRequest,
+  res: NextApiResponse<ApiResponse<ApplicationStageData>>,
+) {
+  const paramValidationError = validatePostRequest(req);
+  if (paramValidationError !== null) {
+    return res.status(HttpStatus.BAD_REQUEST).json(createJsonResponse({}, messages[paramValidationError]));
+  }
+
+  const { type, date } = req.body;
+  const parameters: ApplicationStagePostData = { type, date: new Date(date) };
+
+  const newStage: ApplicationStageData = await prisma.applicationStage.create({
+    data: { ...parameters, applicationId: application.id },
+    select: { applicationId: true, id: true, type: true, date: true, emojiUnicodeHex: true, remark: true },
+  });
+  return res.status(HttpStatus.CREATED).json(createJsonResponse(newStage, messages[MessageType.CREATED]));
 }
+
+async function getHandler(
+  application: ApplicationDataWithStages,
+  res: NextApiResponse<ApiResponse<ApplicationDataWithStages>>,
+) {
+  return res.status(HttpStatus.OK).json(createJsonResponse(application, messages[MessageType.CREATED]));
+}
+
+/** Validates path parameters and body. */
+function validatePathParameters(req: NextApiRequest): Nullable<MessageType> {
+  const applicationId = req.query.applicationId as string;
+  return Number.isInteger(applicationId) ? MessageType.INVALID_APPLICATION_ID : null;
+}
+
+/** Validates path parameters and body. */
+function validatePostRequest(req: NextApiRequest): Nullable<MessageType> {
+  const { type, date } = req.body;
+
+  if (typeof type != 'string' || !(type in ApplicationStageType)) {
```

Comment:
Is using `!=` here better than `!==`. I always that we always use `!==` for strict equality.

Reply:
I wasn't really concerned because typeof should always return a string, a non-strict equals will work the same way but I can fix this for standardisation. 

Could we add this to the es lint rules too?